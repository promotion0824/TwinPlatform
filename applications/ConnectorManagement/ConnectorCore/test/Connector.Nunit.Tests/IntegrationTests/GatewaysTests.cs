namespace Connector.Nunit.Tests.IntegrationTests
{
    using System.Collections.Generic;
    using System.Linq;
    using System.Net;
    using System.Threading.Tasks;
    using Connector.Nunit.Tests.Infrastructure.Extensions;
    using Connector.Nunit.Tests.TestData;
    using ConnectorCore.Entities;
    using FluentAssertions;
    using NUnit.Framework;
    using Snapshooter.NUnit;

    public class GatewaysTests
    {
        [Test]
        public async Task GetGatewaysForSite_ReturnsGateways()
        {
            using var client = IntegrationFixture.Server.CreateClientRandomUser();
            var gateways = await client.GetJsonAsync<List<GatewayEntity>>($"sites/{Constants.SiteIdDefault}/gateways");

            var sorted = gateways
                .Select(g =>
                {
                    g.Connectors = g.Connectors.OrderBy(i => i.Name).ToList();
                    return g;
                })
                .OrderBy(i => i.Id);

            sorted.MatchSnapshot(options => options.IgnoreAllFields("LastHeartbeatTime").IgnoreAllFields("LastUpdatedAt"));
        }

        [Test]
        public async Task GetGatewayById_ReturnsGateway()
        {
            using var client = IntegrationFixture.Server.CreateClientRandomUser();
            var gateway = await client.GetJsonAsync<GatewayEntity>($"sites/{Constants.SiteIdDefault}/gateways/{Constants.GatewayId1}");
            gateway.MatchSnapshot(options => options.IgnoreAllFields("LastHeartbeatTime").IgnoreAllFields("LastUpdatedAt"));
        }

        [Test]
        public async Task GetGatewayById_WrongFormatGuid_Returns400()
        {
            using var client = IntegrationFixture.Server.CreateClientRandomUser();
            var response = await client.GetAsync($"sites/{Constants.SiteIdDefault}/gateways/360d2af2-446b-4b1f-bb6f-20af553ea28");
            response.IsSuccessStatusCode.Should().BeFalse();
            response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
        }

        [Test]
        public async Task GetGatewayById_WrongGuid_Returns404()
        {
            using var client = IntegrationFixture.Server.CreateClientRandomUser();
            var response = await client.GetAsync($"sites/{Constants.SiteIdDefault}/gateways/360d2af2-446b-4b1f-bb6f-20af553ea282");
            response.IsSuccessStatusCode.Should().BeFalse();
            response.StatusCode.Should().Be(HttpStatusCode.NotFound);
        }
    }
}
