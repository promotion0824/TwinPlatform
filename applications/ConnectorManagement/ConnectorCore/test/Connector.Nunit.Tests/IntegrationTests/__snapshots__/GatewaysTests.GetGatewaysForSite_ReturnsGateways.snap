[
  {
    "Id": "05390110-5599-4c67-8a73-65efcf595d5c",
    "Name": "Gateway 4",
    "CustomerId": "8b242df4-e0fa-4c15-99b4-66597a8d56ae",
    "SiteId": "6a8cb6ef-f23b-4608-a08b-0b779fd616cb",
    "Host": "abc123",
    "IsEnabled": false,
    "LastHeartbeatTime": "2022-02-28T17:56:28.736Z",
    "IsOnline": true,
    "Connectors": []
  },
  {
    "Id": "7cce88da-6649-4637-872c-7f0481fbaf1c",
    "Name": "Gateway 1",
    "CustomerId": "8b242df4-e0fa-4c15-99b4-66597a8d56ae",
    "SiteId": "6a8cb6ef-f23b-4608-a08b-0b779fd616cb",
    "Host": "abc123",
    "IsEnabled": true,
    "LastHeartbeatTime": "2022-02-28T17:56:28.736Z",
    "IsOnline": true,
    "Connectors": [
      {
        "Id": "360d2af2-446b-4b1f-bb6f-20af553ea289",
        "Name": "Connector 1",
        "ClientId": "8b242df4-e0fa-4c15-99b4-66597a8d56ae",
        "SiteId": "6a8cb6ef-f23b-4608-a08b-0b779fd616cb",
        "Configuration": "",
        "ConnectorTypeId": "5113b3bf-0de6-4a56-818e-7828f938b6fa",
        "ErrorThreshold": 5,
        "IsEnabled": true,
        "IsLoggingEnabled": true,
        "RegistrationId": "0e1e581e-b47f-432d-bfae-d334eb0a3ded",
        "RegistrationKey": "91301a50-b347-4846-a6c8-6a060a7079df",
        "LastImport": null,
        "LastUpdatedAt": "2022-02-28T17:56:28.658Z",
        "ConnectionType": "",
        "PointsCount": 0,
        "IsArchived": false
      },
      {
        "Id": "116bdce3-4ad1-48ce-9585-4adefa894efa",
        "Name": "Connector 2",
        "ClientId": "8b242df4-e0fa-4c15-99b4-66597a8d56ae",
        "SiteId": "6a8cb6ef-f23b-4608-a08b-0b779fd616cb",
        "Configuration": "",
        "ConnectorTypeId": "5113b3bf-0de6-4a56-818e-7828f938b6fa",
        "ErrorThreshold": 0,
        "IsEnabled": true,
        "IsLoggingEnabled": true,
        "RegistrationId": null,
        "RegistrationKey": null,
        "LastImport": null,
        "LastUpdatedAt": "2022-02-28T17:56:28.659Z",
        "ConnectionType": "",
        "PointsCount": 0,
        "IsArchived": false
      }
    ]
  },
  {
    "Id": "8979ecbd-c95b-42da-8c64-99ddc28c5426",
    "Name": "Gateway 2",
    "CustomerId": "8b242df4-e0fa-4c15-99b4-66597a8d56ae",
    "SiteId": "6a8cb6ef-f23b-4608-a08b-0b779fd616cb",
    "Host": "abc123",
    "IsEnabled": true,
    "LastHeartbeatTime": "2022-02-28T17:56:28.736Z",
    "IsOnline": true,
    "Connectors": [
      {
        "Id": "a11ee966-6e2b-4a7c-b300-bcaa04a5fb94",
        "Name": "Connector to be deleted",
        "ClientId": "8b242df4-e0fa-4c15-99b4-66597a8d56ae",
        "SiteId": "6a8cb6ef-f23b-4608-a08b-0b779fd616cb",
        "Configuration": "",
        "ConnectorTypeId": "5113b3bf-0de6-4a56-818e-7828f938b6fa",
        "ErrorThreshold": 0,
        "IsEnabled": true,
        "IsLoggingEnabled": true,
        "RegistrationId": null,
        "RegistrationKey": null,
        "LastImport": null,
        "LastUpdatedAt": "2022-02-28T17:56:28.66Z",
        "ConnectionType": "",
        "PointsCount": 0,
        "IsArchived": false
      },
      {
        "Id": "1dd7684e-562f-4afa-904f-7ef4903c1b85",
        "Name": "Connector to be deleted 2",
        "ClientId": "8b242df4-e0fa-4c15-99b4-66597a8d56ae",
        "SiteId": "6a8cb6ef-f23b-4608-a08b-0b779fd616cb",
        "Configuration": "",
        "ConnectorTypeId": "5113b3bf-0de6-4a56-818e-7828f938b6fa",
        "ErrorThreshold": 0,
        "IsEnabled": true,
        "IsLoggingEnabled": true,
        "RegistrationId": null,
        "RegistrationKey": null,
        "LastImport": null,
        "LastUpdatedAt": "2022-02-28T17:56:28.66Z",
        "ConnectionType": "",
        "PointsCount": 0,
        "IsArchived": false
      }
    ]
  },
  {
    "Id": "ae8c13b2-6e4e-4b2e-99a1-6b84787a5094",
    "Name": "Gateway 3",
    "CustomerId": "8b242df4-e0fa-4c15-99b4-66597a8d56ae",
    "SiteId": "6a8cb6ef-f23b-4608-a08b-0b779fd616cb",
    "Host": "abc123",
    "IsEnabled": false,
    "LastHeartbeatTime": "2022-02-28T17:56:28.736Z",
    "IsOnline": true,
    "Connectors": [
      {
        "Id": "2207dd6c-228f-4cad-86fb-bac7af5800b7",
        "Name": "Connector 5",
        "ClientId": "8b242df4-e0fa-4c15-99b4-66597a8d56ae",
        "SiteId": "6a8cb6ef-f23b-4608-a08b-0b779fd616cb",
        "Configuration": "",
        "ConnectorTypeId": "856a76da-e8e0-4f65-ae2b-f14a087725a2",
        "ErrorThreshold": 5,
        "IsEnabled": true,
        "IsLoggingEnabled": true,
        "RegistrationId": null,
        "RegistrationKey": null,
        "LastImport": null,
        "LastUpdatedAt": "2022-02-28T17:56:28.66Z",
        "ConnectionType": "",
        "PointsCount": 0,
        "IsArchived": false
      }
    ]
  }
]
